Map correct

alexpattison opened the pull request:
Moved logic from routes.js into controller.js

alexpattison committed:
Move calculateAndDisplayRoute into services.js

alexpattison committed:
Refactor to allow calculateAndDisplayRoute to have access to $scope from inputsController

alexpattison committed:
Implement working refactor for the map route

Currently Justin's marker functions won't work because I added the map view directly into the main.html, so it isn't hitting the map controller anymore.

alexpattison committed:
Move marker logic into controller

The marker logic is currently broken. It seems like the problem is that it is trying to set the markers before the restaurants have populated.

alexpattison committed:
Fix marker logic

Moved it all into the controller

jbg46209 commented on src/client/services/services.js:
```diff
@@ -66,6 +66,24 @@ app.factory('RestaurantAndRoute', ['$http', function($http) {
+
+    calculateAndDisplayRoute: (directionsService, directionsDisplay, start, end) => {
+      directionsService.route({
+        // hardcoded the important route from where Eric and I live to school
```
Is this still hard coded?

jbg46209 commented on src/client/routes.js:
```diff
@@ -27,52 +27,13 @@ app.config(['$stateProvider', '$urlRouterProvider', function($stateProvider, $ur
-          //add restaurant markers
-          RestaurantAndRoute.addMarkers(map);
-        }
+        // controller: function($scope, RestaurantAndRoute) {
```
Seems to be added to controller.js, therefore this block of commented out code should be removed

jbg46209 commented on src/client/views/main.html:
```diff
@@ -7,8 +7,14 @@
+    <div id='map'>
+    </div>
   </div>
+  <!-- <div ui-view='map'>
```
Remove if no longer being used.

alexpattison replied:
We're planning on refactoring back to this, so I'm going to leave it. This is just a temporary solution because of issues with being able to access $scope.

jbg46209 commented:
Looks good. Just clean up some commented out code.

ZolayvarE commented:
I'm having a little bit of difficulty parsing the code. I think it needs more comments.

cchrispy commented:
Could you add some comments describing what's going on, especially in the google.maps stuff. It'd make it easier to understand when we look back to it later in the week.

alexpattison committed:
Refactor code in controller

This commit removes the initial coord from the map element, and changes the function signature of the init map function to take no parameters. Added comments to services and controller.

AlexPattison commented:
Cool. I took care of it.
